src/component/vocabulary/VocabularyEdit.tsx

```diff
@@ -57,6 +61,10 @@ export class VocabularyEdit extends React.Component<
   };
 
   public onSave = () => {
+    const modifiedDocument = Object.assign({}, this.props.vocabulary.document, {
```

Comment:
Could you also add a check for change in the document label so that we do not update the document when there is no change to it?